### PR TITLE
Dango Day Easter Egg

### PR DESCRIPTION
**Pull request by TheColaber**

### CODE REVIEW

File: addons/dango-rain/addon.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+  ],
+  "userstyles": [
+    {
+      "url": "style.css",
+      "matches": ["https://scratch.mit.edu/users/*"]
+    }
+  ],
+  "tags": ["easterEgg"],
+  "enabledByDefault": true,
```
Comment (WorldLanguages) [inline addons/dango-rain/addon.json:38]: Note: this is intended

**Comment by jeffalo** [comment]
is this visibile in the settings page without konami code? i'd imagine many users want it off and it shouldn't be hidden behind the konami code.

File: webpages/settings/index.js [hunk 2]
```diff
@@ -362,6 +362,15 @@ const vue = (window.vue = new Vue({
       const matchesEasterEgg = addonManifest.tags.includes("easterEgg")
         ? this.selectedTab === "easterEgg" || addonManifest._enabled
         : true;
+
+      // April fools
+      if (addonManifest._addonId === "dango-rain") {
+        const now = new Date().getTime() / 1000;
+        if (this.selectedTab === "easterEgg") {
+        } else if (now < 1617364800 && now > 1617192000) {
+        } else return false;
```
Comment (RedGuy12) [inline webpages/settings/index.js:371]: why the empty `if`

Comment (TheColaber) [reply webpages/settings/index.js:371]: this is a wl thing but it just means a "not" for the conditions... not a big deal...

File: addons/dango-rain/addon.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "Dango rain on profiles",
+  "description": "Profiles that include \"dango\" on their \"About Me\" section will have dangos raining on the sides of the screen!",
+  "info": [
+    {
+      "type": "notice",
+      "text": "If you disable this addon, you will only be able to find it if you enter the konami code.",
```
Comment (RedGuy12) [inline addons/dango-rain/addon.json:7]: I don't think we want to tell them how to get the easter eggs addons...

File: addons/dango-rain/userscript.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
+let dangorain = false,
```
Comment (RedGuy12) [inline addons/dango-rain/userscript.js:1]: Aren't these global now? What am I missing?

File: addons/dango-rain/userscript.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  }
+};
+addEventListener("resize", () => {
+  setEltWidth(dangoContainerLeft);
+  setEltWidth(dangoContainerRight);
+});
+
+export default async function ({ addon, global, console, msg }) {
+  const notifClose = Object.assign(document.createElement("span"), {
+    style: `
```
Comment (RedGuy12) [inline addons/dango-rain/userscript.js:68]: Move to css file?

File: addons/dango-rain/userscript.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
+let dangorain = false,
+  dangoContainerLeft,
+  dangoContainerRight,
+  vissiableDangos,
```
Comment (RedGuy12) [inline addons/dango-rain/userscript.js:4]: Typo

File: addons/dango-rain/style.css [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  height: 100%;
+  position: fixed;
+  top: 0;
+  right: 0;
+}
+.sa-dango-notice {
+  bottom: 20px;
+  position: fixed;
+  z-index: 999;
+  background: #000;
```
Comment (RedGuy12) [inline addons/dango-rain/style.css:18]: Does/should this change in dark mode?

**Review by RedGuy12 — CHANGES_REQUESTED** [review]

File: addons/dango-rain/addon.json (hunk 3 above)
Comment (TheColaber) [reply addons/dango-rain/addon.json:7]: 1. this is on the todo
2. we actually want users to know about the addons... why wouldn't we...

File: addons/dango-rain/userscript.js (hunk 4 above)
Comment (TheColaber) [reply addons/dango-rain/userscript.js:1]: No?? window.* makes things global.

File: addons/dango-rain/style.css (hunk 7 above)
Comment (TheColaber) [reply addons/dango-rain/style.css:18]: No looks, fine. same as sa notification

File: addons/dango-rain/userscript.js (hunk 5 above)
Comment (TheColaber) [reply addons/dango-rain/userscript.js:68]: this is taken from cs.js, and also code by W_L... up to him

File: webpages/settings/index.js [hunk 8]
```diff
@@ -362,6 +362,15 @@ const vue = (window.vue = new Vue({
       const matchesEasterEgg = addonManifest.tags.includes("easterEgg")
         ? this.selectedTab === "easterEgg" || addonManifest._enabled
         : true;
+
+      // April fools
+      if (addonManifest._addonId === "dango-rain") {
+        const now = new Date().getTime() / 1000;
+        if (this.selectedTab === "easterEgg") {
+        } else if (now < 1617364800 && now > 1617192000) {
```
Comment (RedGuy12) [inline webpages/settings/index.js:370]: The `now < 1617364800 && ` will hide the addon behind the komoni code after April fools even if it is enabled. Remove it pls.

Comment (TheColaber) [reply webpages/settings/index.js:370]: this was intended by W_L.

Comment (RedGuy12) [reply webpages/settings/index.js:370]: Could we show it if they have force after aprill fools?

File: webpages/settings/index.js [hunk 9]
```diff
@@ -173,7 +173,7 @@ const deserializeSettings = async (str, manifests, confirmElem) => {
       });
     }
     await syncSet({
-      globalTheme: !!obj.core.lightTheme,
+      globalTheme: !!obj.core?.lightTheme,
```
Comment (WorldLanguages) [inline webpages/settings/index.js:176]: Does not belong on this PR

File: addons/dango-rain/userscript.js (hunk 4 above)
Comment (RedGuy12) [reply addons/dango-rain/userscript.js:1]: but it's not in a function so it inherits global scope right?

File: addons/dango-rain/userscript.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
+let dangorain = false, dangoContainerLeft, dangoContainerRight, vissiableDangos, noticeTimeout;
+const div = () => document.createElement("div");
+const noticeText = div();
+noticeText.className = "sa-dango-notice";
+// TODO: localize
+noticeText.innerText = `The dangos filled on this page are visible because this user had the word "dango" in their "About Me"WIWO section of their profile.
+If you would like to do the same to your profile, just include "dango" in there too!
+You can always disable this addon on the settings page!
+Happy April Fools Day from the ScratchAddon developers!`;
```
Comment (GrahamSH-LLK) [inline addons/dango-rain/userscript.js]: Scratch Addons

Comment (TheColaber) [reply addons/dango-rain/userscript.js]: outdated bruh stop

File: addons/dango-rain/style.css [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+  left: 20px;
+  border-radius: 10px;
+  padding: 5px;
+}
+
+.sa-dango {
+  color: #fff;
+  font-size: 3em;
+  font-family: Arial, sans-serif;
+  text-shadow: 0 0 5px #000;
```
Comment (GrahamSH-LLK) [inline addons/dango-rain/style.css:31]: This is unnecessary.

Comment (TheColaber) [reply addons/dango-rain/style.css:31]: maybe without any themes...

Comment (GrahamSH-LLK) [reply addons/dango-rain/style.css:31]: No, this is completely useless. There is no longer any text in the `::before`s. That styling isn't doing anything.

File: addons/dango-rain/userscript.js (hunk 4 above)
Comment (WorldLanguages) [reply addons/dango-rain/userscript.js:1]: @RedGuy12 Module context does not pollute the global scope implicitly, unlike non-module contexts.

**Pull request merged**